src/homos.c

```diff
@@ -292,23 +293,28 @@ homo_hook_collect(void* user_param, uint16_t const nr, uint16_t const* map) {
 
 static void get_automorphism_group_from_gap(Obj digraph_obj, PermColl* out) {
   DIGRAPHS_ASSERT(CALL_1ARGS(IsDigraph, digraph_obj) == True);
+  if (CALL_1ARGS(IsMultiDigraph, digraph_obj) == True) {
+    ErrorQuit("expected a digraph without multiple edges!", 0L, 0L);
+  }
   Obj o = CALL_1ARGS(AutomorphismGroup, digraph_obj);
   o     = CALL_1ARGS(GeneratorsOfGroup, o);
   DIGRAPHS_ASSERT(IS_LIST(o));
   clear_perm_coll(out);
   out->degree = PERM_DEGREE;
   DIGRAPHS_ASSERT(out->capacity >= LEN_LIST(o));
+  Int k = 0;  // perm coll index
   for (Int i = 1; i <= LEN_LIST(o); ++i) {
     DIGRAPHS_ASSERT(ISB_LIST(o, i));
     DIGRAPHS_ASSERT(IS_PERM2(ELM_LIST(o, i)) || IS_PERM4(ELM_LIST(o, i)));
     Obj p = ELM_LIST(o, i);
     DIGRAPHS_ASSERT(LargestMovedPointPerm(p) <= PERM_DEGREE);
     if (LargestMovedPointPerm(p) == 0) {
+      // the index k must be separate from the index i because of this continue
```

Comment:
Is `k` supposed to be modified by this loop? Because as far as I can tell it isn't, and so I don't see the point of `k` because below, you're always doing:
```
Perm q = out->perms[0];
out->size++;
DIGRAPHS_ASSERT(out->size == 1);
```

Reply:
Thanks @wilfwilson `k` is modified though, in line 315.

Reply:
Duh, silly me. My internal C parser is broken...

I was thinking that the `}` brace on line 314 was ending the `for` loop, which meant that line 315 was outside the loop! I must be tired...